### PR TITLE
HSEARCH-4435 Custom Elasticsearch settings/mapping require the backend to be started

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4435

### CODE REVIEW

File: integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/bootstrap/ElasticsearchBootstrapIT.java [hunk 1]
```diff
@@ -270,6 +276,49 @@ public void noVersionCheck_versionOverrideOnStart_compatibleVersion() {
 		assertThat( elasticsearchClientSpy.getRequestCount() ).isNotZero();
 	}
 
+	/**
+	 * Check custom settings/mapping can be parsed when version_check.enabled is false,
+	 * meaning the elasticsearch link is only established *after*
+	 * the custom settings/mapping are parsed.
+	 */
+	@Test
+	@TestForIssue(jiraKey = {"HSEARCH-4435"})
```
Comment (fax4ever) [inline integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/bootstrap/ElasticsearchBootstrapIT.java:285]: a minor stuff: usually we don't put the curly brackets around the jira key

**Review by fax4ever — APPROVED** [review]
LGMT

File: integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/bootstrap/ElasticsearchBootstrapIT.java (hunk 1 above)
Comment (yrodiere) [reply integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/bootstrap/ElasticsearchBootstrapIT.java:285]: Right, probably some IDE auto-complete gone wrong. Well, I'll be more careful next time; no need to bother CI with such a change.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**